python_src/src/lamp_py/migrations/versions/performance_manager/43153d536c2a_feed_info_init_date_trigger.py

```diff
@@ -0,0 +1,67 @@
+"""feed_info init_date trigger
+
+Revision ID: 43153d536c2a
+Revises: 98aa70293578
+Create Date: 2023-04-25 06:53:09.672206
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "43153d536c2a"
+down_revision = "98aa70293578"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    op.add_column(
+        "static_feed_info",
+        sa.Column("version_pub_date", sa.Integer(), nullable=True),
+    )
+
+    op.create_index(
+        op.f("ix_static_feed_info_version_pub_date"),
+        "static_feed_info",
+        ["version_pub_date"],
+        unique=False,
+    )
+
+    create_feed_info_insert_function = """
+        CREATE OR REPLACE FUNCTION insert_feed_info() RETURNS TRIGGER AS $$ 
+        BEGIN 
+            IF NEW.feed_version ~ '\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+\d{2}:\d{2}' THEN
+                NEW.version_pub_date := replace((substring(NEW.feed_version from '\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+\d{2}:\d{2}')::timestamptz at time zone 'US/Eastern')::date::text,'-','')::integer;
+            ELSE
+                NEW.version_pub_date := NEW.feed_start_date;
+            END IF;
+            RETURN NEW;
```

Comment:
This function checks for a the regex match first, to make sure the `substring` function will not fail.

If there is no regex match, then it just inserts `feed_start_date` as `version_pub_date`